layouts/partials/footer.html

```diff
@@ -4,9 +4,17 @@
     {{ .Site.Params.CustomFooterHTML | safeHTML }}
     {{ end }}
 
-    {{ if .Site.Params.footer }}
-    <span>&copy; {{ now.Year }} {{ .Site.Params.Footer}}</span>
+    {{/* Print a year range if .Date and .Lastmod are present */}}
+    {{ if and (.Date) (.Lastmod) (.Site.Params.Footer) (eq .Kind "page") (gt (time.Format "2006" .Lastmod) (time.Format "2006" .Date)) }}
+    <span>&copy; {{ time.Format "2006" .Date }}-{{ time.Format "2006" .Lastmod }} {{ .Site.Params.Footer }} {{ .Site.Copyright }}</span>
+    {{/* Use .Date as the copyright year */}}
+    {{ else if and (.Date) (.Site.Params.Footer) (eq .Kind "page") }}
+        <span>&copy; {{ time.Format "2006" .Date }} {{ .Site.Params.Footer }} {{ .Site.Copyright }}</span>
+    {{/* Use the current year for other Kinds, and pages without Date or Lastmod */}}
+    {{ else if (.Site.Params.Footer) }}
+        <span>&copy; {{ now.Year }} {{ .Site.Params.Footer }}</span>
```

Comment:
Can you change the logic flow to
```template
{{ footerText := "&copy; now.year" }}
// logic for setting copyRight in footerText

{{ if .params.footer}}
  footerText += " {{footer}""
  
<span>{{ footertext }}</span>
```

Reply:
Thanks @yashmehrotra,

I tried something similar to your syntax, but `+=` is not a supported operator. [Using variables](https://gohugo.io/templates/partial/#variable-scoping) didn't provide much joy either.

First example (code fails):

```
    {{ $footerText := "" }}

    {{/* Print a year range if .Date and .Lastmod are present */}}
    {{ if and (.Date) (.Lastmod) (.Site.Params.Footer) (eq .Kind "page") (gt (time.Format "2006" .Lastmod) (time.Format "2006" .Date)) }}
        {{ $footerText := "&copy; {{ time.Format '2006' .Date }}-{{ time.Format '2006' .Lastmod }} {{ .Site.Params.Footer }} {{ .Site.Copyright }}" }}

    {{/* Use .Date as the copyright year */}}
    {{ else if and (.Date) (.Site.Params.Footer) (eq .Kind "page") }}
        {{ $footerText := "&copy; {{ time.Format '2006' .Date }} {{ .Site.Params.Footer }} {{ .Site.Copyright }}" }}

    {{/* Use the current year for other Kinds, and pages without Date or Lastmod */}}
    {{ else if (.Site.Params.Footer) }}
        {{ $footerText := "&copy; {{ now.Year }} {{ .Site.Params.Footer }}" }}
    {{ end }}

    <span>{{ $footerText }}</span>
```

After some reading, I tried [Scratch](https://gohugo.io/methods/page/scratch/#methods), however this produces even longer code. `.Scratch.Add` also suffers with arbitrary characters such as a space `" "` or dash `"-"`.

Second example (code works, but there is no formatting. See below screenshot):

```
    {{ .Scratch.Set "footerText" "" }}

    {{/* Print a year range if .Date and .Lastmod are present */}}
    {{ if and (.Date) (.Lastmod) (.Site.Params.Footer) (eq .Kind "page") (gt (time.Format "2006" .Lastmod) (time.Format "2006" .Date)) }}
        {{ .Scratch.Add "footerText" (.Date | time.Format "2006") }}
        {{ .Scratch.Add "footerText" (.Lastmod | time.Format "2006") }}
        {{ .Scratch.Add "footerText" .Site.Params.Footer }}
        {{ .Scratch.Add "footerText" .Site.Copyright }}

    {{/* Use .Date as the copyright year */}}
    {{ else if and (.Date) (.Site.Params.Footer) (eq .Kind "page") }}
        {{ .Scratch.Add "footerText" (.Date | time.Format "2006") }}
        {{ .Scratch.Add "footerText" .Site.Params.Footer }}
        {{ .Scratch.Add "footerText" .Site.Copyright }}

    {{/* Use the current year for other Kinds, and pages without Date or Lastmod */}}
    {{ else if (.Site.Params.Footer) }}
        {{ .Scratch.Add "footerText" (time.Now | time.Format "2006") }}
        {{ .Scratch.Add "footerText" .Site.Params.Footer }}
    {{ end }}
```

![image](https://github.com/user-attachments/assets/d90cf256-00a6-471c-a4b5-ace30b6fd2e4)

N.B. Don't ask why `.Scratch.Add` is happier with invoking `time.Format` after a bar character. I don't know either :grin: 

More research required. Just keeping you up-to-date.

Reply:
6268dd9 After some trial and error, I believe I've compressed this down as much as possible. Hope this is what you intended :+1: